tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,206 @@
+import pytest
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.callbacks import AverageModelCheckpoint
+from tensorflow_addons.optimizers import MovingAverage
+import os
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+TRAIN_SAMPLES = 10
+NUM_CLASSES = 2
+INPUT_DIM = 3
+NUM_HIDDEN = 5
+BATCH_SIZE = 5
+EPOCHS = 5
+test_model_filepath = "test_model.h5"
+
+moving_avg = MovingAverage(
+    tf.keras.optimizers.SGD(lr=2.0), sequential_update=True, average_decay=0.5,
+)
+
+
+def get_data_and_model(optimizer=moving_avg):
+    x = tf.random.normal([TRAIN_SAMPLES, INPUT_DIM])
+    y = tf.random.normal([TRAIN_SAMPLES, NUM_CLASSES])
+    inputs = keras.layers.Input(INPUT_DIM)
+    layer1 = keras.layers.Dense(NUM_HIDDEN, input_dim=INPUT_DIM, activation="relu")(
+        inputs
+    )
+    outputs = keras.layers.Dense(NUM_CLASSES, activation="softmax")(layer1)
+    model = keras.Model(inputs=inputs, outputs=outputs)
+    model.compile(loss="categorical_crossentropy", optimizer=optimizer, metrics=["acc"])
+    return x, y, model
+
+
+def test_compatibility_with_some_opts_only():
+    x, y, model = get_data_and_model(optimizer="rmsprop")
+    avg_model_ckpt = AverageModelCheckpoint(
+        update_weights=True, filepath=test_model_filepath
+    )
+    with pytest.raises(
+        TypeError,
+        match="AverageModelCheckpoint is only used when trainingwith"
+        + " MovingAverage or StochasticAverage",
+    ):
+        assert model.fit(
```

Comment:
```suggestion
        model.fit(
```

That would try to convert the keras history into a boolean object. Tests don't necesarily need assert statements.